test/qpu_interface.jl

```diff
@@ -975,7 +1178,12 @@ end
 
         qpu = QPU(test_client, expected_project_id, status_request_throttle = no_throttle)
 
-        histogram = transpile_and_run_job(qpu, circuit, shot_count)
+        histogram = transpile_and_run_job(
+            qpu,
+            circuit,
+            shot_count;
+            transpiler = get_transpiler(qpu; connectivity = connectivity),
```

Comment:
See above comment regarding get_transpiler. Also, this is something we can take as an argument to the constructor of AnyonQPU.

Reply:
as discussed offline, having an optional argument `connectivity` when constructing QPUs conflicts with the purpose of `get_metadata()`, which will clobber anything present